homeassistant/components/mealie/todo.py

```diff
@@ -0,0 +1,260 @@
+                f"An error occurred adding an item to {self._shopping_list.name}"
+            ) from exception
+
+        await self.coordinator.async_refresh()
+
+    async def async_update_todo_item(self, item: TodoItem) -> None:
+        """Update an item on the list."""
+        list_items = self.coordinator.shopping_list_items[self._shopping_list.list_id]
+
+        for items in list_items:
+            if items.item_id == item.uid:
+                position = items.position
+                break
+
+        list_items = self.coordinator.shopping_list_items[self._shopping_list.list_id]
+
+        list_item: ShoppingItem | None = next(
+            (x for x in list_items if x.item_id == item.uid), None
+        )
+
+        if list_item:
+            udpdate_shopping_item = MutateShoppingItem(
+                item_id=list_item.item_id,
+                list_id=list_item.list_id,
+                note=list_item.note,
+                display=list_item.display,
+                checked=item.status == TodoItemStatus.COMPLETED,
+                position=list_item.position,
+                is_food=list_item.is_food,
+                disable_amount=list_item.disable_amount,
+                quantity=list_item.quantity,
+                label_id=list_item.label_id,
+                food_id=list_item.food_id,
+                unit_id=list_item.unit_id,
+            )
+
+            stripped_item_summary = (
+                item.summary.strip() if item.summary else item.summary
+            )
+
+            if list_item.display.strip() != stripped_item_summary:
+                udpdate_shopping_item.note = stripped_item_summary
+                udpdate_shopping_item.position = position
+                udpdate_shopping_item.is_food = False
+                udpdate_shopping_item.food_id = None
+                udpdate_shopping_item.quantity = 0.0
+                udpdate_shopping_item.checked = item.status == TodoItemStatus.COMPLETED
+
+            try:
+                await self.coordinator.client.update_shopping_item(
+                    list_item.item_id, udpdate_shopping_item
+                )
+            except MealieError as exception:
+                raise HomeAssistantError(
+                    f"An error occurred updated an item on {self._shopping_list.name}"
+                ) from exception
+
+            await self.coordinator.async_refresh()
+            return
+
+        raise HomeAssistantError(f"Item {item.uid} not found in shopping list")
+
+    async def async_delete_todo_items(self, uids: list[str]) -> None:
+        """Delete items from the list."""
+        try:
+            for uid in uids:
+                await self.coordinator.client.delete_shopping_item(uid)
+        except MealieError as exception:
+            raise HomeAssistantError(
+                f"An error occurred deleting item(s) on {self._shopping_list.name}"
+            ) from exception
+
+        await self.coordinator.async_refresh()
```

Comment:
```suggestion
        try:
            for uid in uids:
                await self.coordinator.client.delete_shopping_item(uid)
        except MealieError as exception:
            raise HomeAssistantError(
                f"An error occurred deleting item(s) on {self._shopping_list.name}"
            ) from exception
        finally:
            await self.coordinator.async_refresh()
```
So if the item was already deleted, causing the error, we just refresh again to make sure our data is up to date